src/main/java/io/supertokens/oauth/exceptions/OAuthClientException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *    Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.oauth.exceptions;
+
+import java.io.Serial;
+
+public class OAuthClientException extends  OAuthException{
```

Comment:
The name of this exception does not really indicate whats going wrong. Does it mean the client already exists, or client does not exist, or something else?

Reply:
That's more like a generic exception to hold the errors returned by hydra. It's called ..ClientException because it gets thrown when using the client APIs. If we want to use specifically named exceptions (instead of the error message in the `error` and `errorDescription` fields to identify the problem) then we have to discover/know all the possible errors hydra can return and map them to our internal exceptions. I don't think that's feasible. What do you think?

Reply:
Okay, after todays discussion, this was renamed.